### PR TITLE
#7 Setting up test 

### LINKED ISSUE #7: Setup tests
The aim is to make unit tests runnable

Topic: https://github.com/OpenPaaS-Suite/esn/issues/11

Sample PR: https://github.com/OpenPaaS-Suite/esn-frontend-calendar/pull/25

### PR DESCRIPTION
**Pull request by pbson**
image

### CODE REVIEW

File: src/app/app.module.js [hunk 1]
```diff
@@ -23,4 +25,4 @@ require('./invitation-message/blue-bar/invitation-message-blue-bar.controller.js
 require('./invitation-message/indicator/invitation-message-indicator.component.js');
 require('./resource-management/blue-bar/resource-management-blue-bar.component.js');
 require('./resource-management/blue-bar/resource-management-blue-bar.controller.js');
-require('./resource-management/indicator/resource-management-indicator.component.js');
+require('./resource-management/indicator/resource-management-indicator.component.js');
```
Comment (tuanlc) [inline src/app/app.module.js]: Add blank line before

File: test/fixtures/logger-noop.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ *
+ * @return {{log: Function, warn: Function, error: Function, debug: Function, info: Function}}
+ */
+module.exports = {
+  log: noop,
+  warn: noop,
+  error: noop,
+  debug: noop,
+  info: noop
+};
```
Comment (tuanlc) [inline test/fixtures/logger-noop.js]: Do we need this one?

File: webpack.test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  },
+  plugins: [
+    new webpack.NormalModuleReplacementPlugin(
+      /node_modules\/esn-frontend-common-libs\/src\/frontend\/js\/modules\/i18n\/i18n-loader.service.js/,
+      i18nLoaderMockPath
+    ),
+    new webpack.ProvidePlugin({
+      chai: chaiPath,
+      _: lodashPath,
+      moment: momentPath
```
Comment (tuanlc) [inline webpack.test.js]: Do we need moment for this repo?

File: webpack.test.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    })
+  ],
+  module: {
+    rules: [
+      // We must ignore all *.run.js files
```
Comment (tuanlc) [inline webpack.test.js]: is it true?

**Review by tuanlc — COMMENTED** [review]
Please check all the `src/fixtures` files to see if we need all of them

File: src/app/app.module.js [hunk 5]
```diff
@@ -7,20 +7,23 @@ angular.module('esn.inbox-calendar', [
...
 require('esn-frontend-calendar/src/esn.calendar.libs/app/app.module.js');
 require('esn-frontend-calendar/src/esn.resource.libs/app/app.module.js');
+require('esn-frontend-common-libs/src/frontend/js/modules/lodash-wrapper.js');
 
 require('./app.config');
 require('./app.constants');
 require('./invitation-message/blue-bar/invitation-message-blue-bar.component.js');
 require('./invitation-message/blue-bar/invitation-message-blue-bar.controller.js');
 require('./invitation-message/indicator/invitation-message-indicator.component.js');
+
```
Comment (tuanlc) [inline src/app/app.module.js]: Why?

Comment (pbson) [reply src/app/app.module.js]: @tuanlc Did you ask me to add a blank line `before`?

Comment (tuanlc) [reply src/app/app.module.js]: Yes, but the blank line at the end of the file but not here

**Review by tuanlc — DISMISSED** [review]

File: package.json [hunk 6]
```diff
@@ -17,5 +18,46 @@
...
+    "exports-loader": "^1.0.0",
+    "expose-loader": "^1.0.0",
+    "favicons-webpack-plugin": "^3.0.1",
+    "file-loader": "^6.0.0",
+    "glob-all": "^3.2.1",
+    "html-loader": "^1.1.0",
+    "html-webpack-plugin": "^4.3.0",
+    "ignore-loader": "^0.1.2",
+    "karma": "4.2.0",
+    "karma-coverage": "1.1.2",
```
Comment (itswillta) [inline package.json]: :trollface:

File: package.json [hunk 7]
```diff
@@ -17,5 +18,46 @@
     "fullcalendar": "3.9.0",
     "lodash": "2.4.1"
   },
-  "devDependencies": {}
+  "devDependencies": {
+    "angular-mocks": "1.3.20",
+    "chai": "3.0.0",
+    "chai-datetime": "1.4.0",
+    "chai-shallow-deep-equal": "1.4.0",
+    "clean-webpack-plugin": "^3.0.0",
```
Comment (itswillta) [inline package.json]: :trollface:

File: package.json [hunk 8]
```diff
@@ -17,5 +18,46 @@
...
+    "karma-rawfixtures-preprocessor": "linagora/karma-rawfixtures-preprocessor",
+    "karma-sinon-chai": "^2.0.2",
+    "karma-spec-reporter": "0.0.26",
+    "karma-webpack": "^4.0.2",
+    "less-loader": "^6.1.1",
+    "mocha": "6.2.0",
+    "pug": "^3.0.0",
+    "pug-html-loader": "^1.1.5",
+    "pug-loader": "^2.4.0",
+    "replace-in-file": "^6.1.0",
```
Comment (itswillta) [inline package.json]: Where is this used?

File: src/app/app.module.js [hunk 9]
```diff
@@ -7,14 +7,16 @@ angular.module('esn.inbox-calendar', [
   'esn.calendar.libs',
   'esn.resource.libs',
   'linagora.esn.videoconference',
-  'linagora.esn.videoconference.calendar'
+  'linagora.esn.videoconference.calendar',
+  'esn.lodash-wrapper'
```
Comment (itswillta) [inline src/app/app.module.js]: We don't use use this anymore. If you see lodash injected as an AngularJS dependency, remove it.

**Review by itswillta — CHANGES_REQUESTED** [review]

File: package.json (hunk 6 above)
Comment (pbson) [reply package.json]: Fixed

File: package.json (hunk 7 above)
Comment (pbson) [reply package.json]: Fixed

File: package.json (hunk 8 above)
Comment (pbson) [reply package.json]: Fixed

File: src/app/app.module.js (hunk 9 above)
Comment (pbson) [reply src/app/app.module.js]: Fixed

**Review by itswillta — APPROVED** [review]

**Review by tuanlc — APPROVED** [review]

**Pull request merged**